app/services/filter_builder.rb

```diff
@@ -0,0 +1,31 @@
+class FilterBuilder
+  attr_accessor :klass, :params, :results
+
+  def initialize(object_type, params = nil)
```

Comment:
Is it not possible to pass in `FilterBuilder.new(Product, params)?

Reply:
That's how it works, by passing in the object type and the params. I made it optional to pass in the params, but that may not be the best idea.

Reply:
Ah I understand

Reply:
I just removed that option so a FilterBuilder must be passed params. This is probably the best way since you can't do any filtering without params.